rules/prebuilt_framework.bzl

```diff
@@ -0,0 +1,93 @@
+load("@build_bazel_rules_apple//apple/internal:apple_framework_import.bzl", "AppleFrameworkImportInfo")
+load("@build_bazel_rules_swift//swift/internal:providers.bzl", "SwiftUsageInfo")
+
+def apple_dynamic_framework_import_patched(name, **kwargs):
+    """Patches an apple_dynamic_framework_import target based on the problems reported in https://github.com/bazel-ios/rules_ios/issues/55
+
+    Args: same as the ones of apple_dynamic_framework_import
+    """
+
+    visibility = kwargs.pop("visibility", None)
+    tags = kwargs.pop("tags", [])
+
+    legacy_target_label = "_" + name
+    apple_dynamic_framework_import(
+        name = legacy_target_label,
+        tags = tags + ["manual"],
+        **kwargs
+    )
+
+    target_patcher(
+        name = name,
+        legacy_target = legacy_target_label,
+        isDynamic = True,
+        visibility = visibility,
+        tags = tags,
+    )
+
+def apple_static_framework_import_patched(name, **kwargs):
+    """Patches an apple_static_framework_import target based on the problems reported in https://github.com/bazel-ios/rules_ios/issues/55
+
+    Args: same as the ones of apple_static_framework_import
+    """
+
+    visibility = kwargs.pop("visibility", None)
+    tags = kwargs.get("tags", [])
+
+    legacy_target_label = "_" + name
+    apple_static_framework_import(
+        name = legacy_target_label,
+        tags = tags + ["manual"],
+        **kwargs
+    )
+
+    target_patcher(
+        name = name,
+        legacy_target = legacy_target_label,
+        isDynamic = False,
+        visibility = visibility,
+        tags = tags,
+    )
+
+def _target_patcher_impl(ctx):
+    old_objc_provider = ctx.attr.legacy_target[apple_common.Objc]
+
+    # Merge providers
+    objc_provider_fields = {
+        "providers": [old_objc_provider],
+        # Adding the module_map to the headers.
+        # This "hack" is just a way to propagate the module map. swift_rules has a field in its provider
+        # called swiftc_inputs where you can pass extra dependencies to the compiler. Unfortunately
+        # the objc provider has no such field. The "hack" is to pass the module map as a header so that
+        # it makes it to the list of dependencies. Since it doesn't end in .h it gets ignored but it's
+        # still propagated as a dependency. This "hack" is based on:
+        # a) https://github.com/bazelbuild/rules_apple/blob/313eeb838497e230f01a7367ae4555aaf0cac62e/apple/internal/apple_framework_import.bzl#L98
+        # b) https://github.com/bazel-ios/rules_ios/blob/1755c694f8d5cc1ed256bba7ccf122a3fbc4addf/rules/framework.bzl#L247
+        "header": old_objc_provider.module_map,
+    }
+    new_objc_provider = apple_common.new_objc_provider(**objc_provider_fields)
+
+    # I can not find a way to iterate on the existing providers, so instead I list here the keys to all of them
+    # (you can see the keys for the existing providers of a target by just printing the target)
```

Comment:
@ob any idea about how to iterate the existing providers and avoid hardcoding them here? I feel like maybe this providers change depending on the binary...

Reply:
Does `dir` on the target work?

Reply:
Nope, calling `dir(ctx.attr.legacy_target)` prints the following: 
```
["actions", "data_runfiles", "default_runfiles", "files", "files_to_run", "label", "output_groups"]
```